### PR TITLE
Finished the function for tidying the data

### PR DESCRIPTION
**Pull request by samzzzzzh**
Please take a look, any feedback is appreciated!

### CODE REVIEW

File: R/data_cleaning.R [hunk 1]
```diff
@@ -0,0 +1,25 @@
+#' Creates a new data frame with selected columns dropped from an old dataFrame
```
Comment (a-kong) [inline R/data_cleaning.R:1]: - dataFrame should be changed to data frame
- replace "an" with "the"

Comment (samzzzzzh) [reply R/data_cleaning.R:1]: Good catch! I will fix this.

File: R/data_cleaning.R [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+#' @param columns Name(s) of column(s) from the dataFrame 
+#'
+#' @return A new data frame with columns dropped from the old data frame
+#'
+#' @export
+#'
+#' @examples
+#' data_cleaning(fire_data, c("X","Y","month"))
+data_cleaning <- function(dataFrame, columns){
+  if(!is.data.frame(dataFrame)){
```
Comment (a-kong) [inline R/data_cleaning.R:13]:
```suggestion
  if (!is.data.frame(dataFrame)) {
```

File: R/data_cleaning.R (hunk 1 above)
Comment (Jaskaran1116) [reply R/data_cleaning.R:1]: looks good to me (code wise)

**Review by Jaskaran1116 — APPROVED** [review]

File: R/data_cleaning.R [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    stop("DataFrame must be a data frame! Please try again.")
+  }
+  colNames = names(dataFrame)
+  for(c in columns){
```
Comment (a-kong) [inline R/data_cleaning.R:17]:
```suggestion
  for (c in columns) {
```

File: R/data_cleaning.R [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    if({c} %in% colNames){
```
Comment (a-kong) [inline R/data_cleaning.R:18]:
```suggestion
    if ({c} %in% colNames) {
```

File: R/data_cleaning.R [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+      dataFrame = dataFrame |> select(-{c})
+    }else{
```
Comment (a-kong) [inline R/data_cleaning.R:20]:
```suggestion
    } else {
```

File: tests/testthat/test-data_cleaning.R [hunk 6]
```diff
@@ -0,0 +1,21 @@
+library(testthat)
+
+test_that("data_cleaning should report an error when a dataFrame is not inputed", {
```
Comment (a-kong) [inline tests/testthat/test-data_cleaning.R:3]:
```suggestion
test_that("data_cleaning should report an error when a data frame is not inputed", {
```

File: tests/testthat/test-data_cleaning.R [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  expect_error(data_cleaning(1,"people"))
+  expect_error(data_cleaning("dataFrame", "gender"))
+  expect_error(data_cleaning(FALSE, "num"))
+})
+
+test_that("data_cleaning should report an error when columns names are incorrect",{
```
Comment (a-kong) [inline tests/testthat/test-data_cleaning.R:9]:
```suggestion
test_that("data_cleaning should report an error when columns names are incorrect", {
```

File: tests/testthat/test-data_cleaning.R [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+  expect_error(data_cleaning(test_data, c("people","number")))
+  expect_error(data_cleaning(test_data, c("PeoPLE")))
+  expect_error(data_cleaning(test_data, c("")))
+})
+
+test_that("data_cleaning should output a dataFrame with only two columns (excluding people)",{
```
Comment (a-kong) [inline tests/testthat/test-data_cleaning.R:15]:
```suggestion
test_that("data_cleaning should output a data frame with only two columns (excluding people)", {
```

File: tests/testthat/test-data_cleaning.R [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+  expect_equivalent(data_cleaning(test_data, "people"), test_data_two_columns)
+})
+
+test_that("data_cleaning should output a dataFrame with only one columns (only gender)",{
```
Comment (a-kong) [inline tests/testthat/test-data_cleaning.R:19]:
```suggestion
test_that("data_cleaning should output a data frame with only one columns (only gender)", {
```

File: tests/testthat/helper-data_cleaning.R [hunk 10]
```diff
@@ -0,0 +1,7 @@
+people <- c("john" , "zeddy", "mike")
+gender <- c("male", "male", 'female')
+num <- c(1:3)
+
+test_data <- data.frame(people, gender, num)
+test_data_two_columns <- data.frame(gender,num)
```
Comment (a-kong) [inline tests/testthat/helper-data_cleaning.R:6]:
```suggestion
test_data_two_columns <- data.frame(gender, num)
```

**Review by a-kong — APPROVED** [review]
added some comments, but other than that it looks good!

**Pull request merged**